src/components/DocSearch/DocSearch.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ *
+ * @emails react-core
+ */
+
+import React, {Component} from 'react';
+import {colors, media} from 'theme';
+
+class DocSearch extends Component {
+  state = {
+    disabled: false,
+  };
+  componentDidMount() {
+    // Initialize Algolia search.
+    // TODO Is this expensive? Should it be deferred until a user is about to search?
+    // eslint-disable-next-line no-undef
```

Comment:
Might be nice but might also be unnecessary. Feel free to leave it for a follow-up, or if you're interested in maybe doing it- measure the perf first to see if it's actually worth doing?

Reply:
Maybe not, I just thought of removing the complexity from the Template and just let the DocSearch do everything, mostly because of its own setState behavior.